File: Programming_Basics_with_Python/conditional_statements_advanced_exercise/08._on_time_for_the_exam.py
hour_of_examine = int(input())
minutes_of_examine = int(input())
hour_of_arriving = int(input())
minutes_of_arriving = int(input())
time_of_examine = hour_of_examine * 60 + minutes_of_examine
time_of_arriving = hour_of_arriving * 60 + minutes_of_arriving
if time_of_arriving > time_of_examine:
    print("Late")
elif time_of_examine - 30 <= time_of_arriving <= time_of_examine:
    print("On time")
else:
    print("Early")
if time_of_examine - 60 < time_of_arriving < time_of_examine:
    print(f"{time_of_examine - time_of_arriving} minutes before the start")
elif time_of_arriving <= time_of_examine - 60:
    difference = time_of_examine - time_of_arriving
    hours = difference // 60
    minutes = difference % 60
    print(f"{hours}:{minutes:0>2d} hours before the start")
elif time_of_examine < time_of_arriving < time_of_examine + 60:
    print(f"{time_of_arriving - time_of_examine} minutes after the start")
elif time_of_arriving >= time_of_examine + 60:
    difference = time_of_arriving - time_of_examine
    hours = difference // 60
    minutes = difference % 60
    print(f"{hours}:{minutes:0>2d} hours after the start")

File: Programing_Basics_with_JS/conditionalStatementsAdvancedExercise/08.OnTimeForTheExam.js
function onTimeForTheExam(input) {
    let examHours = Number(input[0]);
    let examMinutes = Number(input[1]);
    let arrivalHours = Number(input[2]);
    let arrivalMinutes = Number(input[3]);

    let examTime = examHours * 60 + examMinutes;
    let arrivalTime = arrivalHours * 60 + arrivalMinutes;
    let timeDiff = examTime - arrivalTime;
    let timeDiffHour = Math.trunc(timeDiff / 60);
    let timeDiffMinutes = Math.abs(timeDiff) % 60;

    if (arrivalTime > examTime) {
        console.log('Late');

        if (Math.abs(timeDiff) <= 59) {
            console.log(`${Math.abs(timeDiff)} minutes after the start`);
        } else if (Math.abs(timeDiff) >= 60 && timeDiffMinutes < 10) {
            console.log(`${Math.abs(timeDiffHour)}:0${timeDiffMinutes} hours after the start`);
        }   else {
            console.log(`${Math.abs(timeDiffHour)}:${timeDiffMinutes} hours after the start`);
        }

    } else if (timeDiff === 0 || timeDiff <= 30) {
        console.log('On time');
        if (timeDiff !== 0) {
            console.log(`${timeDiff} minutes before the start`);
        }

    } else if (timeDiff > 30) {
        console.log('Early');

        if (timeDiff <= 59) {
            console.log(`${timeDiff} minutes before the start`);
        } else if (timeDiff >= 60 && timeDiffMinutes < 10) {
            console.log(`${timeDiffHour}:0${timeDiffMinutes} hours before the start`);
        } else {
            console.log(`${timeDiffHour}:${timeDiffMinutes} hours before the start`);
        }
    }
}

onTimeForTheExam(["9",
"30",
"9",
"50"])

onTimeForTheExam(["9",
"00",
"8",
"30"])

onTimeForTheExam(["16",
"00",
"15",
"00"])

onTimeForTheExam(["9",
"00",
"10",
"30"])

onTimeForTheExam(["14",
"00",
"13",
"55"])

onTimeForTheExam(["11",
"30",
"8",
"12"])

onTimeForTheExam(["10",
"00",
"10",
"00"])

onTimeForTheExam(["11",
"30",
"10",
"55"])

onTimeForTheExam(["11",
"30",
"12",
"29"])

